ApeWorX/ape
# refactor!: move transaction status and type to ape-ethereum

BREAKING CHANGE: Moves `TransactionStatusEnum` and `TransactionType` from `ape.api.transactions` to `ape_ethereum.transactions`.

### What I did

Right now, the API assumes you use Ethereum-esque transaction status numbers and types.
This PR lessens those assumptions by counting on primitive values in the API and moves the Enum classes to the `ape-ethereum` core plugin.

One weird part is the `Web3Provider` mixin still needs to check the transaction type in an Ethereum context (`Web3Provider` is EVM by nature so this is okay I think).

### How I did it

* Create a `ape_ethereum.transactions` module that contains the transaction implementations, the receipt, and now the enums that were originally in `ape.api`

### How to verify it

You can still make transactions!
I am going to leave this as a draft to ensure it works in the Starknet land.

### Checklist
